Client/Frontend/Sync/BraveCore/Bookmarks/BookmarkAPIExtensions.swift

```diff
@@ -0,0 +1,255 @@
+// Copyright 2021 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import Data
+import BraveCore
+import BraveShared
+import CoreData
+import Shared
+
+private let log = Logger.browserLogger
+
+extension BraveBookmarksAPI {
+
+    // MARK: Internal
+    
+    // Returns the last visited folder
+    // If no folder was visited, returns the mobile bookmarks folder
+    // If the root folder was visited, returns nil
+    func lastVisitedFolder() -> BookmarkNode? {
+        guard Preferences.General.showLastVisitedBookmarksFolder.value,
+              let nodeId = Preferences.Chromium.lastBookmarksFolderNodeId.value else {
+            // Default folder is the mobile node..
+            if let mobileNode = mobileNode {
+                return mobileNode
+            }
+            return nil
```

Comment:
No need to if-let here, can just `return mobileNode`

Reply:
Fix at https://github.com/brave/brave-ios/commit/f64450d0fa36146b6576ce9cd1c874d21f992ae9